### PR TITLE
#84: Stop Asynchronous Server

### PR DESCRIPTION
**Pull request by rahulraghu94**
New helper function to help stop Asynchronous server are being added.
Calling this helper function will kill the main thread

### CODE REVIEW

File: pymodbus/server/async.py [hunk 1]
```diff
@@ -274,9 +274,23 @@ def StartSerialServer(context, identity=None,
...
     reactor.run()
 
+
+def _stop_server(is_mainthread=True):
+    from twisted.internet import reactor
+    if _is_main_thread():
+        reactor.stop()
+    else:
+        reactor.callFromThread(reactor.stop)
+    _logger.debug("Stopping main thread")
```
Comment (dhoomakethu) [inline pymodbus/server/async.py]: @rahulraghu94 This log message will be displayed irrespective of whether the current thread is main or child.

**Comment by rahulraghu94** [comment]
@dhoomakethu , changed the base to `dev`

File: pymodbus/server/async.py [hunk 2]
```diff
@@ -274,9 +274,23 @@ def StartSerialServer(context, identity=None,
     SerialPort(protocol, port, reactor, baudrate)
     reactor.run()
 
+
+def _stop_server(is_mainthread=True):
```
Comment (dhoomakethu) [inline pymodbus/server/async.py]: THe param `is_mainthread=True` is not being used, remove it if unnecessary

File: pymodbus/server/async.py (hunk 1 above)
Comment (rahulraghu94) [reply pymodbus/server/async.py]: Isn't that expected behaviour?

Comment (rahulraghu94) [reply pymodbus/server/async.py]: Oh, sorry. I'm on some meds 📦 
Fixing...

**Pull request merged**